Program.cs

```diff
@@ -15,252 +15,273 @@ public static void printsContentsRefrigerator(List<Refrigerator> allfrige)
                     foreach (Item item in shelf._itemsOnShelf)
                     {
-                        Console.WriteLine("  "+item.ToString()); 
+                        Console.WriteLine("  " + item.ToString());
                     }
                 }
 
             }
         }
-        public static Item.Kashrot checkKosherInput()
+        public static void CheckingIntegrityInput(int userInput)
         {
-
-            int kashrot; ;
-            while (true)
+            if (userInput == -1)
             {
-                Console.WriteLine("Please select the kosher of the item:");
-                Console.WriteLine("1. " + (Item.Kashrot)0);
-                Console.WriteLine("2. " + (Item.Kashrot)1);
-                Console.WriteLine("3. " + (Item.Kashrot)2);
-                string input = Console.ReadLine();
-                if (int.TryParse(input, out kashrot) && kashrot > 0 && kashrot < 4)
-                {
-                    return (Item.Kashrot)(--kashrot);
-
-                }
-                else { Console.WriteLine("Please,enter valid input"); }
+                throw new Exception("Invalid input");
             }
+
         }
-        public static Item.TypeItem checkTypeItemInput()
+        public static int GetKosherInput()
+        {
+            int kashrot;
+            string userInput;
+            Console.WriteLine("Pease select the Kashrot of the item:");
+            Console.WriteLine("1. " + (Item.Kashrot)0);
+            Console.WriteLine("2. " + (Item.Kashrot)1);
+            Console.WriteLine("3. " + (Item.Kashrot)2);
+            userInput = Console.ReadLine();
+            if (!(int.TryParse(userInput, out kashrot)) || kashrot < 1 || kashrot > 3)
+                return -1;
+            return --kashrot;
+
+        }
+        public static int GetTypeItemInput()
         {
             int typeItem;
-            while (true)
-            {
-                Console.WriteLine("lease select the Type of the item:");
-                Console.WriteLine("1. " + (Item.TypeItem)0);
-                Console.WriteLine("2. " + (Item.TypeItem)1);
-                string inp = Console.ReadLine();
-                if (int.TryParse(inp, out typeItem) && typeItem > 0 && typeItem < 3)
-                {
-                    return (Item.TypeItem)(--typeItem);
-                }
-                else { Console.WriteLine("Please,enter valid input"); }
-            }
+            string userInput;
+            Console.WriteLine("Pease select the Type of the item:");
+            Console.WriteLine("1. " + (Item.TypeItem)0);
+            Console.WriteLine("2. " + (Item.TypeItem)1);
+            userInput = Console.ReadLine();
+            if (!(int.TryParse(userInput, out typeItem)) || typeItem < 1 || typeItem > 2)
+                return -1;
+            return --typeItem;
         }
-        public static String checkNameItemInput()
+        public static String GetNameItemInput()
```

Comment:
avoid static method in such an exercise like this.
the main purpose of static is to exposure the method to all layers of the code.